### PR TITLE
upSample class in layers_v2

### PR DESCRIPTION
**Pull request by utkarsh8028**

### CODE REVIEW

File: joey/layers_v2.py [hunk 1]
```diff
@@ -391,3 +391,158 @@ def __init__(self, kernel_size, input_size,
...
+                   layer.result_gradients[dims[0],
+                                          kernel_dims[0], dims[2],
+                                          dims[3]] *
+                   layer.input[dims[0], kernel_dims[1],
+                               kernel_dims[2] + dims[2],
+                               kernel_dims[3] + dims[3]])]
+
+        _, _, height, width = layer.kernel.shape
+
+        if next_layer is not None:
```
Comment (navjotk) [inline joey/layers_v2.py:512]: I don't like the fact that we follow different code paths depending on whether the layer is stand-alone or as part of a larger network.

Comment (utkarsh8028) [reply joey/layers_v2.py:512]: This PR is only for forward propogation.
We have diff equation for back-prop but with same if statement, I continued the original flow for back-propogation as its additional code if you have a previous layer which makes sense.
from what I understand, its not about single or part of network, its whether its the first layer or not.
We can't calculate next_layer variables if next layer is none 
Do you have something else in mind?

Comment (navjotk) [reply joey/layers_v2.py:512]: Following a discussion, we agree that this will go away when we separate layer specification vs layer execution.

Comment (navjotk) [reply joey/layers_v2.py:512]: Issue #17

**Review by navjotk — APPROVED** [review]

**Pull request merged**